app/src/main/java/org/simple/clinic/introvideoscreen/IntroVideoScreen.kt

```diff
@@ -0,0 +1,121 @@
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.util.AttributeSet
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.cast
+import kotlinx.android.synthetic.main.screen_intro_video.view.*
+import org.simple.clinic.R
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.di.injector
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.platform.crash.CrashReporter
+import org.simple.clinic.registration.register.RegistrationLoadingScreenKey
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.util.unsafeLazy
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Named
+
+class IntroVideoScreen(
+    context: Context,
+    attrs: AttributeSet
+) : ConstraintLayout(context, attrs), UiActions {
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @field:[Inject Named("training_video_youtube_id")]
+  lateinit var youTubeVideoId: String
+
+  @Inject
+  lateinit var introVideoEffectHandler: IntroVideoEffectHandler.Factory
+
+  @Inject
+  lateinit var crashReporter: CrashReporter
+
+  private val events: Observable<IntroVideoEvent> by unsafeLazy {
+    Observable
+        .mergeArray(
+            videoClicks(),
+            skipClicks()
+        )
+        .compose(ReportAnalyticsEvents())
+        .cast<IntroVideoEvent>()
+  }
+
+  private val mobiusDelegate by unsafeLazy {
+    MobiusDelegate.forView(
+        events,
+        IntroVideoModel.default(),
+        IntroVideoUpdate(),
+        introVideoEffectHandler.create(this).build()
+    )
+  }
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) return
+
+    context.injector<IntroVideoScreenInjector>().inject(this)
+
+    // Hard-coding to show this simple video view exists because, as of now,
+    // we are not sure if we will have variations of this training video.
+    // We should make the title, duration and video thumbnail configurable in order to improve this.
+    introVideoSubtitle.text = resources.getString(R.string.simple_video_duration, "5:07")
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    mobiusDelegate.start()
+  }
+
+  override fun onDetachedFromWindow() {
+    super.onDetachedFromWindow()
+    mobiusDelegate.stop()
+  }
+
+  override fun openVideo() {
+    openYoutubeLinkForSimpleVideo()
+  }
+
+  override fun openHome() {
+    screenRouter.push(RegistrationLoadingScreenKey())
+  }
+
+  @Suppress("UNCHECKED_CAST")
+  private fun videoClicks(): Observable<IntroVideoEvent> {
+    val clicksFromVideoImage = introVideoImageView.clicks().map { VideoClicked }
+    val clicksFromWatchVideoButton = watchVideoButton.clicks().map { VideoClicked }
+
+    return clicksFromVideoImage.mergeWith(clicksFromWatchVideoButton) as Observable<IntroVideoEvent>
+  }
```

Comment:
The idiomatic way in Rx to cast is using the `.cast` operator. Can we switch to using that instead?

```kotlin
  private fun videoClicks(): Observable<IntroVideoEvent> {
    val clicksFromVideoImage = introVideoImageView.clicks().map { VideoClicked }
    val clicksFromWatchVideoButton = watchVideoButton.clicks().map { VideoClicked }

    return clicksFromVideoImage
        .mergeWith(clicksFromWatchVideoButton)
        .cast()
  }
```

You might need to import the `cast()` extension function to get the one that does not need a `Class`  parameter.